Add a Jenkinsfile for the new dockerized Jenkins instance

mpol opened the pull request.

mpol committed:
Add a Jenkinsfile for the new dockerized Jenkins instance

It should do the same as the build set up in the old Jenkins plus
run Sonar if on the master branch.

Based on the metrics-dlivery-endpoint Jenkinsfile.dockerized:
- use a `pollSCM` trigger because there is no way to setup the github
  webhook yet,
- take the Sonar token from Jenkins credentials,
- use a Maven settings file `deploy-settings.xml` to pass
  the Artifactory credentials from Jenkins to Maven,
- use separate stages for things to do for all branches (build, test)
  and things to do only on master (Sonar, Artifactory).

mpol committed:
Use https for Artifactory access in distributionManagement

With http there are errors with the maven deploy step.

marekhorst commented on Jenkinsfile:
```diff
@@ -0,0 +1,74 @@
+    agent any
+
+    triggers {
+        cron('H H(20-23) * * 1-5')
```
I think primer is quite "stable" in terms of code changes so we might drop cron trigger as well.

mpol replied:
ok

marekhorst commented on Jenkinsfile:
```diff
@@ -0,0 +1,74 @@
+                }
+            }
+            steps {
+		sh 'mvn -B clean package -Dmaven.test.failure.ignore'
```
Is the `maven.test.failure.ignore` just a temporary flag? Or was it propagated from the current build definition?

mpol replied:
This is the way I found to distinguish between failing and unstable (tests failing) builds. The non-pipeline Jenkins Maven projects do this automatically somehow, here we let the maven step succeed and then the junit step will mark the build as unstable. We may not actually need this failing/unstable distinction, but Jenkins has it since forever.

Maybe https://plugins.jenkins.io/pipeline-maven/ would handle this automatically, but it is not installed and there are posts (eg. on stackoverflow) that suggests that people still need to use maven.test.failure.ignore with this plugin.

marekhorst reviewed (approved):
Just made two comments: one question and one idea. Approving, but if you consider those comments as valid then you can introduce the changes before merging.

mpol committed:
Remove the cron trigger from the Jenkinsfile

It was suggested during review that we may get rid of periodic builds
when there are no changes, especially because this repository
is rather "stable" in terms of code changes and dependencies.
